docs/CONTRIBUTING.md

```diff
@@ -23,12 +23,14 @@ sections of this document are linked below:
   - [Design Documentation](#design-documentation)
   - [System Requirements](#system-requirements)
   - [Getting the Sources](#getting-the-sources)
-  - [Getting Set Up \(Rust\)](#getting-set-up-rust)
-  - [Getting Set Up \(JVM\)](#getting-set-up-jvm)
-  - [Getting Set Up \(Documentation\)](#getting-set-up-documentation)
-  - [Building Enso](#building-enso)
-  - [Testing Enso](#testing-enso)
-  - [Running Enso](#running-enso)
+  - Configuration
+    - [Set Up Rust)](#getting-set-up-rust)
+    - [Set Up Documentation](#getting-set-up-documentation)
+  - Enso Engine CLI
```

Comment:
- separating the changes in 6a7159b3f6c67328de2101675f99f7a969209d58
- into own #13131 
- read the [rendered documentation](https://github.com/enso-org/enso/blob/b653828adf50f9a39cf40578fd67a2a0a2907bf8/docs/CONTRIBUTING.md#building-enso-engine)